R/calcIEAPFU.R

```diff
@@ -0,0 +1,51 @@
+#' IEA database of primary, final and useful energy
+#'
+#' Computes data for the Primary Final Useful (PFU) resolution
+#'
+#' @return IEA data as MAgPIE object aggregated to country level
+#'
+#' @author Antoine Levesque, Robin Krekeler
+#'
+#' @seealso \code{\link{calcOutput}}
+#' @examples
+#' \dontrun{ a <- calcOutput("IEA_PFU")
+#' }
+#'
+#' @importFrom madrat toolGetMapping readSource
+#' @importFrom stats na.omit
+#' @importFrom dplyr %>% select filter
+#' @importFrom tidyr unite
+#' @importFrom magclass dimSums getNames
+#' @importFrom utils read.csv2
+#' @export
+
+calcIEAPFU <- function() {
+
+  mapping <- toolGetMapping(type = "sectoral",
+                            name = "structuremappingFE_PFU.csv",
+                            returnPathOnly = TRUE) %>%
+    read.csv2(stringsAsFactors = FALSE, na.strings = "") %>%
+    select("iea_product", "iea_flows", "pfu") %>%
+    na.omit()
+
+  # read in data and convert from ktoe to EJ
+  data <- readSource("IEA", subtype = "EnergyBalances") * 4.1868E-5
```

Comment:
I prefer magic numbers in a separate, well-named variable, but again just my personal preference. Explaining with a comment is definitely also a good way. By the way, what unit is `ktoe`?
```suggestion
  ktoeToEJ <- 4.1868E-5 # explain ktoe?
  data <- readSource("IEA", subtype = "EnergyBalances") * ktoeToEJ
```